pkg/clusterconfig/clusterconfig.go

```diff
@@ -92,13 +102,59 @@ func Get() (*Config, error) {
 
 	cfg := &Config{}
 
-	cm, err := client.ConfigMaps(configNamespace).Get(configName, metaapi.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			logrus.Warnf("config-map %s/%s not found", configNamespace, configName)
-			return cfg, nil
+	cm, cmerr := client.ConfigMaps(configNamespace).Get(configName, metaapi.GetOptions{})
+
+	if cmerr != nil && errors.IsNotFound(cmerr) {
+		logrus.Warnf("config-map %s/%s not found, creating", configNamespace, configName)
+		cm = &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: corev1.SchemeGroupVersion.String(),
+				Kind:       "ConfigMap",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configName,
+				Namespace: configNamespace,
+			},
+			Data: make(map[string]string),
+		}
+	}
+
+	awsCreds, err := client.Secrets(kubeSystemNamespace).Get(awsSecretName, metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
+		logrus.Warnf("secret %s/%s not found, skipping aws configuration", kubeSystemNamespace, awsSecretName)
+	} else {
+		cm.Data["storage_type"] = string(regopapi.StorageTypeS3)
+		if v, ok := awsCreds.Data["aws_access_key_id"]; ok {
+			cm.Data["storage_aws_accesskey"] = string(v)
+		}
+		if v, ok := awsCreds.Data["aws_secret_access_key"]; ok {
+			cm.Data["storage_aws_secretkey"] = string(v)
+		}
```

Comment:
this should all be refactored into functions that are specific for dealing with s3 storage configuration.

Reply:
updated in next push